### PR TITLE
[SPARK-39250][BUILD] Upgrade Jackson to 2.13.3

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade Jackson to 2.13.3.

### Why are the changes needed?

Although Spark is not affected, Jackson 2.13.0~2.13.2 has the following regression which affects the user apps.
- https://github.com/FasterXML/jackson-databind/issues/3446

Here is a full release note.
- https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.13.3

### Does this PR introduce _any_ user-facing change?

No. The previous version is not released yet.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @MaxGekk . I set this as a blocker for Apache Spark 3.3.0.
Please review this when you have some time.

**Review by HyukjinKwon — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -170,8 +170,8 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.13.2</fasterxml.jackson.version>
-    <fasterxml.jackson.databind.version>2.13.2.1</fasterxml.jackson.databind.version>
+    <fasterxml.jackson.version>2.13.3</fasterxml.jackson.version>
+    <fasterxml.jackson.databind.version>2.13.3</fasterxml.jackson.databind.version>
```
Comment (pan3793) [inline pom.xml:174]: I add this property because of SPARK-38665, and it can be removed because all jackson jars share same version again.

**Review by MaxGekk — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (dongjoon-hyun) [reply pom.xml:174]: If you look at the history, SPARK-38665 is not the only one did that. `databind` is frequently separated and merged back repeatedly. :) That's the reason why I decided not to remote that property back in this PR. 

```
[SPARK-38665][BUILD] Upgrade jackson due to CVE-2020-36518
[SPARK-33695][BUILD] Upgrade to jackson to 2.10.5 and jackson-databind to 2.10.5.1
[SPARK-28728][BUILD] Bump Jackson Databind to 2.9.9.3
```

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon , @pan3793 , @MaxGekk . Merged to master/3.3 to unblock next RC.

**Comment by MaxGekk** [comment]
After merge of the changes to 3.3, I observe the error:
```
[info] - Hide credentials in show create table *** FAILED *** (34 milliseconds)
[info]   "[0,10000000d5,5420455441455243,62617420454c4241,414e20200a282031,4e4952545320454d,45485420200a2c47,a29544e49204449,726f20474e495355,6568636170612e67,732e6b726170732e,a6362646a2e6c71,20534e4f4954504f,7462642720200a28,203d2027656c6261,45502e5453455427,200a2c27454c504f,6f77737361702720,2a27203d20276472,2a2a2a2a2a2a2a2a,6574636164657228,2720200a2c272964,27203d20276c7275,2a2a2a2a2a2a2a2a,746361646572282a,20200a2c27296465,3d20277265737527,7355747365742720,a29277265]" did not contain "TEST.PEOPLE" (JDBCSuite.scala:1146)
[info]   org.scalatest.exceptions.TestFailedException:
```
Is it just a coincidence?

**Comment by dongjoon-hyun** [comment]
Yes, that failure was irrelevant to this PR, @MaxGekk .
- It was caused via https://github.com/apache/spark/pull/36632
- It was fixed via https://github.com/apache/spark/pull/36637

**Pull request closed without merging**